docs/celo-codebase/protocol/transactions/escrow.md

```diff
@@ -1,25 +1,224 @@
+    }
+    ```
+
+2. converting the `public key` into a `public address` (referred to as the `paymentId` in this context) using [`publicKeyToAddress()`](https://github.com/celo-org/celo-monorepo/blob/6b6ce69fde8f4868b54abd8dd267e5313c3ddedd/packages/sdk/utils/src/address.ts#L38) from [@celo/utils/lib/address](https://github.com/celo-org/celo-monorepo/blob/6b6ce69fde8f4868b54abd8dd267e5313c3ddedd/packages/sdk/utils/src/address.ts).
+
+    ```ts
+    const publicKeyToAddress = (publicKey: string) =>
+        toChecksumAddress(
+            ensureLeading0x(pubToAddress(toBuffer(ensureLeading0x(publicKey)), true).toString('hex'))
+        )
+    ```
+
+### Option 2: Phone number-based proof of identity
+
+Scenario:
+
+- Alice wants pay to Bob, but Bob doesn't have an account yet.
+- The payment is facilitated by using Bob's phone number
+
+[![Flow diagram for phone number-based payment and proof of identity](https://mermaid.ink/img/pako:eNqlVc1q3DAQfpXBl-xSZ0lDlhIfAilNoRRCoT0agmzNrkVsSZXkXUzIsbc-QvtyeZKOLNvr9W7akPpg9PNp5puZb6SHKFccoySy-L1GmeMHwdaGVakE-ljulIHrUuQYFjQzTuRCM-mA-eU7lueq9jML12F4iNSsqVC6T9yjvmGllWGmgc_YHGJzJZ0hvx56Y3OjtgurykNcprI7XSiJHvheZScWvrTT27rK0Bw_8BzZECdZSWW3UDslR4baFMDp6dWVRyWwJYvWxwVO-ZUYstr5AXCFVp44KNiGqEnoXTbY-TJIwZl1Njs_u4jhfLmk39nFfM8PudnLbkLmLDz9-hFQt8ohqA2aKUobsWG0d49NDLrOaHdvzDg3aO3U15DLBO6l2h71NMK8ubiEt--WnvYSFotFgKLk0wC72MYB-hT1LveCkwqefv885vV10Q10dmEOMkyAo0NTCSks6YOVZQNrlGjIuoVZbYVcQxBXEMF8ymtk6oWsYDacmXuVdTrb6yFPstd_AhW7R9t7-mvh490hH4xnT1NpV2hmcxBy1EgkSQ5WYy5WjccZanth0LvwUtaGrIPaUi5sIXQX7CgTPfFdkc99fbvfpMj_nfC2iC9omzYvJa4cqFVoUE8hN9jaH3XhLI2sWEuodRr9S5HPNMHr5Dj20rVRm2s7SXZA7y63A2YV03ZKy19AI8NU8GtHgTvmhJK2LTtdbx-Rtwnn082DXPdkR7IqmagGMUIo2N5NPtSTLvmMWB3jeCjTrXAFN2w7lWk4PbwFB1kgNaDY7NrDtkF3RaYUaaKxMqqaaHefUm8-npR-06rRV0dOW8EVeMTk8TcizDUTvE2okNRfhaCbDrck_yBO3isziqOKGoTA9BY_eINpRM4qTKOEhhxXrC5dGsVhq0CxLlzY829VGqXykWzUmpPVGy7oPYuSFSstxi3gayPzKHGmxh7UPfQd6vEPwz-9JA)](https://mermaid.live/edit#pako:eNqlVc1q3DAQfpXBl-xSZ0lDlhIfAilNoRRCoT0agmzNrkVsSZXkXUzIsbc-QvtyeZKOLNvr9W7akPpg9PNp5puZb6SHKFccoySy-L1GmeMHwdaGVakE-ljulIHrUuQYFjQzTuRCM-mA-eU7lueq9jML12F4iNSsqVC6T9yjvmGllWGmgc_YHGJzJZ0hvx56Y3OjtgurykNcprI7XSiJHvheZScWvrTT27rK0Bw_8BzZECdZSWW3UDslR4baFMDp6dWVRyWwJYvWxwVO-ZUYstr5AXCFVp44KNiGqEnoXTbY-TJIwZl1Njs_u4jhfLmk39nFfM8PudnLbkLmLDz9-hFQt8ohqA2aKUobsWG0d49NDLrOaHdvzDg3aO3U15DLBO6l2h71NMK8ubiEt--WnvYSFotFgKLk0wC72MYB-hT1LveCkwqefv885vV10Q10dmEOMkyAo0NTCSks6YOVZQNrlGjIuoVZbYVcQxBXEMF8ymtk6oWsYDacmXuVdTrb6yFPstd_AhW7R9t7-mvh490hH4xnT1NpV2hmcxBy1EgkSQ5WYy5WjccZanth0LvwUtaGrIPaUi5sIXQX7CgTPfFdkc99fbvfpMj_nfC2iC9omzYvJa4cqFVoUE8hN9jaH3XhLI2sWEuodRr9S5HPNMHr5Dj20rVRm2s7SXZA7y63A2YV03ZKy19AI8NU8GtHgTvmhJK2LTtdbx-Rtwnn082DXPdkR7IqmagGMUIo2N5NPtSTLvmMWB3jeCjTrXAFN2w7lWk4PbwFB1kgNaDY7NrDtkF3RaYUaaKxMqqaaHefUm8-npR-06rRV0dOW8EVeMTk8TcizDUTvE2okNRfhaCbDrck_yBO3isziqOKGoTA9BY_eINpRM4qTKOEhhxXrC5dGsVhq0CxLlzY829VGqXykWzUmpPVGy7oPYuSFSstxi3gayPzKHGmxh7UPfQd6vEPwz-9JA)
```

Comment:
nice diagram! may be helpful to make the distinction between phone number and Identifier though?